test/common/upstream/load_balancer_impl_test.cc

```diff
@@ -71,14 +72,25 @@ class LoadBalancerBaseTest : public LoadBalancerTestBase {
     host_set.runCallbacks({}, {});
   }
 
-  std::vector<uint32_t> getLoadPercentage() {
-    std::vector<uint32_t> ret;
+  template <typename T, typename FUNC>
+  std::vector<T> aggregatePrioritySetsValues(TestLb& lb, FUNC func) {
+    std::vector<T> ret;
+
     for (size_t i = 0; i < priority_set_.host_sets_.size(); ++i) {
-      ret.push_back(lb_.percentageLoad(i));
+      ret.push_back((lb.*func)(i));
```

Comment:
No, it does not work. Compiler tries to call lb's func method and complains that method is not defined.

Reply:
Perhaps just pass an std::function instead of a templatized function? That would definitely be more readable. Up to you.

Reply:
I used template because logic is the same (iterate through priority sets), but type of return value is different: `vector<uint32_t>` for percentageLoad and `vector<bool>` for panic setting.

Reply:
I think you could templatize the return type of std::function. In any case, it's not a big deal either way.